spec/tasks/support/gem_base64_spec.rb

```diff
@@ -0,0 +1,77 @@
+require "tasks/support/gem_base64"
+require "manageiq_performance/version"
+require "fileutils"
+
+def build_gem file
+  pkg_dir = File.dirname file
+  FileUtils.mkdir_p pkg_dir
+  FileUtils.touch file
+  file_obj = File.new(file, "w+")
+
+  Gem::Package.new(file_obj)
+              .tap{ |p| p.spec = GemBase64.gemspec }
+              .build
+end
+
+def decode_and_untar gem_string, pattern="*"
+  # Convert from a base64 string to a tar ball file contents
+  gem_io = StringIO.new Base64.decode64(gem_string)
+  # Un tar the specific file for comparison
+  extracted_contents = ""
+  gem_tar = Gem::Package::TarReader.new(gem_io)
+  gem_tar.each do |entry|
+    next unless entry.full_name == 'data.tar.gz'
+
+    Zlib::GzipReader.wrap entry do |gzio|
+      Gem::Package::TarReader.new(gzio).each do |dot_gem_file|
+        next unless File.fnmatch pattern, dot_gem_file.full_name, File::FNM_DOTMATCH
+        extracted_contents = dot_gem_file.read
+        # don't break here to avoid zlib warning
+      end
+    end
+
+    break # ignore further entries (optimization)
+  end
+
+  extracted_contents
+end
+
+describe GemBase64 do
+  let(:version) { Gem::Version.new(ManageIQPerformance::VERSION) }
+  let(:spec_dir) { File.expand_path "../..", File.dirname(__FILE__) }
```

Comment:
Decided to go with the extra `../` since `__dir__`, I found, is actually a method, and I would prefer to use the constant.  Thanks for the suggestion.